src/server/routes/timeoffRequests.js

```diff
@@ -1,54 +1,81 @@
 const router = require('express').Router();
 const TimeoffRequests = require('../../models/timeoffRequests');
+const { validateDate, validateTime, validateType } = require('../../models/utilities');
 
-//test route
 router.post('/', (request, response) => {
   const timeoffRequest = request.body.text;
-  console.log(timeoffRequest);
-
+  const userID = request.body.user_id;
   let message = {
     "response_type": "ephemeral",
-    "text": "Your timeoff request was recorded!",
+    "text": null,
     "attachments": [
       {
-        "text": timeoffRequest
+        "text": `Your entry was: "${timeoffRequest}"`
       }
     ]
   };
 
-  response.json(message);
-
-});
+  /* EXAMPLE BODY */
+  // { token: 'ifVkOH3RGkzllGo64zWgjyBZ',
+  // team_id: 'T3ZNAQTAP',
+  // team_domain: 'learnersguild',
+  // channel_id: 'D4WL4N36V',
+  // channel_name: 'directmessage',
+  // user_id: 'U4Y1L7RB9',
+  // user_name: 'breyana',
+  // command: '/timeoff',
+  // text: 'this is a message',
+  // response_url: 'https://hooks.slack.com/commands/T3ZNAQTAP/283341594723/oMoIJgwuODf67Mt48NSZiu9q',
+  // trigger_id: '283457877556.135758843363.921f83381be87b9cc374b986db5ef31b' }
 
+  let requestArray = timeoffRequest.split(' ');
+  const requestType = requestArray.shift();
 
-//real route. Not functional until db is created.
-// router.post('/', (request, response) => {
-//   const timeoffRequest = request.body.text;
-//   console.log(timeoffRequest);
-//
-//   const newRequest = {
-//     username,
-//     typeOfTimeoff,
-//     dateOff,
-//     timeStart,
-//     timeEnd
-//   }
-//
-//   let message = {
-//     "response_type": "ephemeral",
-//     "text": "Your timeoff request was recorded!",
-//     "attachments": [
-//       {
-//         "text": timeoffRequest
-//       }
-//     ]
-//   };
-//
-// TimeoffRequests.add(newRequest)
-// .then(result => {
-//   response.json(message);
-// })
-// .catch(error => response.json({error: error.message}));
-// });
+  if (!validateType(requestType)) {
+    message.text = "Make sure your first word is the type of entry: sick, vacation, or overtime";
+  } else {
+    let databaseEntries = [];
```

Comment:
@breyana there is a lot going on in this function. Can you break this function down into smaller functions? Looks like this function is doing a few things. 
1. parsing the message that is coming in from the params
2. Validating the message to ensure it has all the correct params.
3. Inserting the data into the database.

Its good practice to ensure that your `route` function doesnt have any logic in it, and is just delegating to other functions. So if you extract the code into the functions listed above, the code will look a lot cleaner, and you will have a better idea on how to refactor it as well.